contracts/fair-burn/src/contract.rs

```diff
@@ -0,0 +1,454 @@
+use crate::error::ContractError;
+use crate::msg::{ConfigResponse, ExecuteMsg, InstantiateMsg, QueryMsg, SudoMsg};
+use crate::state::{Config, CONFIG};
+
+use cosmwasm_std::{
+    coin, ensure, to_binary, Addr, BankMsg, Binary, Coin, Decimal, Deps, DepsMut, Env, Event,
+    MessageInfo, StdResult, Uint128,
+};
+use cw2::set_contract_version;
+use cw_utils::maybe_addr;
+use sg_std::{create_fund_fairburn_pool_msg, Response, NATIVE_DENOM};
+use std::collections::BTreeMap;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:sg-fair-burn";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let config = Config {
+        fee_percent: Decimal::percent(msg.fee_bps),
```

Comment:
Would be better to do the division (`/ Uint128::from(100u128)`) here so `fee_percent` actually means percent. Submitted a bps conversion PR to CW: https://github.com/CosmWasm/cosmwasm/pull/1715.